### PR TITLE
Improve schema reload performance by pre-filtering joined rows.

### PR DESCRIPTION
**Pull request by arthurschreiber**
## Description

When reloading the database schema, a query to fetch table information and tablespace information is executed. This query can take a long time when run on a mysql instance with many tables across many databases.

We notice the impact of this query during database setup in our CI environments, as this query is executed on each DDL statement (and thus gets incrementally slower the more tables are being created).

Performance of this query was improved before in https://github.com/vitessio/vitess/pull/9632 by pre-filtering the rows of the `information_schema.innodb_sys_tablespaces` before joining them to rows from `information_schema.tables`, but that change was accidentally reverted in https://github.com/vitessio/vitess/pull/10703.

This brings back the pre-filtering.

---

Can this be backported to the 14.x releases? 🙇‍♂️ 

## Related Issue(s)

https://github.com/vitessio/vitess/pull/9632
https://github.com/vitessio/vitess/pull/10703

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Review by systay — APPROVED** [review]

**Comment by glortho** [comment]
👋🏼 Greetings @systay. This is an important change to unblock some of our efforts. Do you have any sense of ETA for it landing in a snapshot release here https://github.com/planetscale/vitess-releases/releases? (Or just a sense of merge ETA?)

**Review by frouioui — APPROVED** [review]
Looks good to me. There is a test using this constant, it is not used otherwise.

**Comment by frouioui** [comment]
@glortho, I am waiting for confirmation regarding the frequency of the planetscale-vitess releases. I will get back to you as soon as I have it.

Otherwise, the PR looks good to me, let's merge it. The `Check Pull Request labels` seems stuck; however, the PR is labeled correctly. I will bypass the branch protection rules.

**Comment by frouioui** [comment]
> I was unable to backport this Pull Request to the following branches: `release-14.0`.

Nothing to backport in this Pull Request, the same change already exist in `release-14.0`: https://github.com/vitessio/vitess/pull/9632

**Comment by glortho** [comment]
@frouioui this PR actually corrects a regression introduced by https://github.com/vitessio/vitess/pull/9632

**Comment by glortho** [comment]
Sorry I was mistaken in a previous comment. Do you mean you would just revert this [change](https://github.com/vitessio/vitess/pull/10703)?

**Comment by frouioui** [comment]
I meant that we don't need to backport/change anything on `release-14.0` as it already contains the changes made in this current Pull Request (#11043). The `TablesWithSize57` variable already has the same value.

**Comment by frouioui** [comment]
@glortho - The frequency of the releases made here (https://github.com/planetscale/vitess-releases/releases) is approximately every 14 days.

**Comment by Anders-PlanetScale** [comment]
@glortho 
I did roll a release today, just now https://github.com/planetscale/vitess-releases/releases/tag/604547f636

NOTE: our release build script is having some issues with cross compile for MacOSX/Darwin so the Darwin package is an empty tar file, if you require Mac binaries you should compile on OSX locally from source for now. (we hope to fix this over the next few days). If you need assistance with that please let us know.

**Comment by glortho** [comment]
Great thank you @Anders-PlanetScale @frouioui!

**Pull request merged**
